cfgov/v1/tests/test_links.py

```diff
@@ -22,6 +25,68 @@ def test_gov_link(self):
         self.assertIn('class="icon-link_text"', output)
 
 
+class LinkTest(TestCase):
+
+    def setUp(self):
+
+        self.page = mock.MagicMock()
+        self.request = mock.MagicMock()
+        self.request.url = 'http://localhost:8000/path/to/page/'
+        self.page.url = 'http://localhost:8000/path/to/some/other/page/'
+        self.page.live = True
+        self.page.specific.shared = True
+        self.context = mock.MagicMock()
+
+        def context_dict(name):
+            return {'request': self.request}[name]
+
+        self.context.__getitem__.side_effect = context_dict
+
+    def test_get_protected_url_no_page(self):
+        """
+        Confirm that in the absence of a page, get_protected_url returns the hash.
+        """
+        result = get_protected_url(self.context, None)
+        self.assertEqual(result, '#')
+
+    def test_get_protected_url_no_url(self):
+        """
+        Confirm that in the absence of a page URL, get_protected_url returns None.
+        """
+        self.page.url = None
+        result = get_protected_url(self.context, self.page)
+        self.assertIsNone(result)
+
+    def test_get_protected_url_live_page(self):
+        """
+        Confirm that if a live page is requested from the same host as the request
+        we get back the original page url.
+        """
+        self.page.url = 'http://localhost:8000/path/to/some/other/page/'
+        result = get_protected_url(self.context, self.page)
+        self.assertEqual(result, self.page.url)
+
+    def test_get_protected_url_non_live_page(self):
+        """
+        Confirm that if a non-live page is request, we get back the hash.
```

Comment:
I might update this to "Confirm that if a non-live page is requested from a live page, we get back the hash" since it's only true under those conditions.